handlers/auth_code_request_create.go

```diff
@@ -76,6 +76,20 @@ func CreateAuthCodeRequest(authCodeReqSvc *service.AuthCodeRequestService) http.
 			}
 		}
 
+		hasFiledWithinPeriod, err := service.CheckCompanyFilingHistory(request.CompanyNumber)
```

Comment:
Should this be moved to check the company filing history _before_ getting the officer details above?

Reply:
Yeah that makes more sense. The query to check the company officers is more logic heavy than this check so if we can avoid doing it we should. 

Moved it to before